File: python/数组/243. 最短单词距离.py
from typing import List
class Solution:
    def shortestDistance(self, words: List[str], word1: str, word2: str) -> int:
        if word1 == word2:
            return 0
        ans = float('+inf')
        for i,word in enumerate(words):
            if word == word1:
                x,y = i-1,i+1
                while x>=0 or y < len(words):
                    if x>=0 and words[x] == word2:
                        ans = min(ans,i-x)
                    if y<len(words) and words[y] == word2:
                        ans = min(ans,y-i)
                    x-=1
                    y+=1
        return ans

File: go/array/243. 最短单词距离.go
package array

import (
	"fmt"
	"math"
)

func shortestDistance(words []string, word1 string, word2 string) int {
	if word1 == word2 {
		return 0
	}
	ans := math.MaxInt32
	for i, word := range words {
		fmt.Println(i, word)
		if word == word1 {
			x, y := i-1, i+1
			for x >= 0 || y < len(words) {
				if x >= 0 && words[x] == word2 {
					if i-x < ans {
						ans = i - x
					}
				}
				if y < len(words) && words[y] == word2 {
					if y-i < ans {
						ans = y - i
					}
				}
				x--
				y++
			}
		}
	}
	return ans
}
